lib/ansible/modules/cloud/azure/azure_rm_containerregistry_facts.py

```diff
@@ -0,0 +1,264 @@
+        required: True
+    name:
+        description:
+            - The name of the container registry.
+    retrieve_credentials:
+        description:
+            - Retrieve credentials for container registry.
+        type: bool
+        default: no
+    tags:
+        description:
+            - Limit results by providing a list of tags. Format tags as 'key' or 'key:value'.
+
+extends_documentation_fragment:
+    - azure
+
+author:
+    - "Zim Kalinowski (@zikalino)"
+
+'''
+
+EXAMPLES = '''
+  - name: Get instance of Registry
+    azure_rm_containerregistry_facts:
+      resource_group: sampleresourcegroup
+      name: sampleregistry
+
+  - name: List instances of Registry
+    azure_rm_containerregistry_facts:
+      resource_group: sampleresourcegroup
+'''
+
+RETURN = '''
+registries:
+    description: A list of dictionaries containing facts for registries.
+    returned: always
+    type: complex
+    contains:
+        id:
+            description:
+                - The resource ID.
+            returned: always
+            type: str
+            sample: "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/myResourceGroup/providers/Microsoft.ContainerRegistry/registr
+                    ies/myRegistry"
+        name:
+            description:
+                - The name of the resource.
+            returned: always
+            type: str
+            sample: myRegistry
+        location:
+            description:
+                - The location of the resource. This cannot be changed after the resource is created.
+            returned: always
+            type: str
+            sample: westus
+        admin_user_enabled:
+            description:
+                - Is admin user enabled.
+            returned: always
+            type: bool
+            sample: yes
+        sku:
+            description:
+                - The SKU name of the container registry.
+            returned: always
+            type: str
+            sample: classic
+        provisioning_state:
+            description:
+                - Provisioning state of the container registry
+            returned: always
+            type: str
+            sample: Succeeded
+        login_server:
+            description:
+                - Login server for the registry.
+            returned: always
+            type: str
+            sample: acrd08521b.azurecr.io
+        credentials:
+            description:
+                - Credentials, fields will be empty if admin user is not enabled for ACR
+            return: always
+            type: complex
+            contains:
+                username:
+                    description:
+                        - The user name for container registry.
+                    returned: when registry exists and C(admin_user_enabled) is set
+                    type: str
+                    sample: zim
+                password:
```

Comment:
jordan mentioned to add no_log to returned value this morning. do you know how to add it?

Reply:
not really, we can search... but I guess I we have option to specifically enable passwords, we can skip it for now

Reply:
don't think so. the option is weak, just make it little difficult to get credential. no_log is really make it not available in another other place other than ansible

Reply:
@yungezz  no_log can be placed in the playbook, we do not have to do anything in the module itself